domain/src/main/java/com/teamagam/gimelgimel/domain/map/SelectEntityInteractor.java

```diff
@@ -3,36 +3,189 @@
+import com.teamagam.gimelgimel.domain.map.entities.interfaces.IGeoEntityVisitor;
+import com.teamagam.gimelgimel.domain.map.entities.mapEntities.AlertEntity;
+import com.teamagam.gimelgimel.domain.map.entities.mapEntities.GeoEntity;
+import com.teamagam.gimelgimel.domain.map.entities.mapEntities.ImageEntity;
+import com.teamagam.gimelgimel.domain.map.entities.mapEntities.MyLocationEntity;
+import com.teamagam.gimelgimel.domain.map.entities.mapEntities.PointEntity;
+import com.teamagam.gimelgimel.domain.map.entities.mapEntities.SensorEntity;
+import com.teamagam.gimelgimel.domain.map.entities.mapEntities.UserEntity;
+import com.teamagam.gimelgimel.domain.map.entities.symbols.MyLocationSymbol;
+import com.teamagam.gimelgimel.domain.map.entities.symbols.PointSymbol;
+import com.teamagam.gimelgimel.domain.map.entities.symbols.UserSymbol;
+import com.teamagam.gimelgimel.domain.map.repository.GeoEntitiesRepository;
+import com.teamagam.gimelgimel.domain.map.repository.SelectedEntityRepository;
+import com.teamagam.gimelgimel.domain.messages.entity.Message;
 import com.teamagam.gimelgimel.domain.messages.repository.EntityMessageMapper;
 import com.teamagam.gimelgimel.domain.messages.repository.MessagesRepository;
 
+import java.util.Arrays;
+
 import rx.Observable;
 
 @AutoFactory
-public class SelectEntityInteractor extends DoInteractor {
+public class SelectEntityInteractor extends BaseDataInteractor {
 
-    private EntityMessageMapper mEntityMessageMapper;
+    private final EntityMessageMapper mEntityMessageMapper;
     private final MessagesRepository mMessagesRepository;
-    private String mEntityId;
+    private final GeoEntitiesRepository mGeoEntitiesRepository;
+    private final SelectedEntityRepository mSelectedEntityRepository;
+    private final String mEntityId;
 
     protected SelectEntityInteractor(
             @Provided ThreadExecutor threadExecutor,
             @Provided EntityMessageMapper entityMessageMapper,
             @Provided MessagesRepository messagesRepository,
+            @Provided GeoEntitiesRepository geoEntitiesRepository,
+            @Provided SelectedEntityRepository selectedEntityRepository,
             String entityId) {
         super(threadExecutor);
         mEntityMessageMapper = entityMessageMapper;
         mMessagesRepository = messagesRepository;
+        mGeoEntitiesRepository = geoEntitiesRepository;
+        mSelectedEntityRepository = selectedEntityRepository;
         mEntityId = entityId;
     }
 
     @Override
-    protected Observable buildUseCaseObservable() {
+    protected Iterable<SubscriptionRequest> buildSubscriptionRequests(
+            DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+        DataSubscriptionRequest selectRelatedMessage =
+                factory.create(createSelectRelatedMessageObservable());
+
+        DataSubscriptionRequest colorSelection = factory.create(
+                Observable.just(mEntityId)
```

Comment:
Maybe extract a method?
Maybe even extract a method including the factory.create and then you get the form-
buildSubscriptionRequests()
{
  first = buildFirstRequest
  second = buildSecondRequest
  return as list(first, second)
}